src/components/contextual/stake/StakePreview.vue

```diff
@@ -49,9 +49,14 @@ const {
   stakeBPT,
   unstakeBPT,
 } = useStaking();
+
+const numSharesToModify =
+  props.action === 'stake'
+    ? balanceFor(getAddress(props.pool.address))
+    : stakedSharesForProvidedPool.value;
```

Comment:
`props.action` and `stakedSharesForProvidedPool.value` are reactive properties, so I think this should be a computed property.